modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/ServiceOrderManager.java

```diff
@@ -117,7 +119,8 @@ public void completeOrderProgress(UUID orderId, TaskStatus taskStatus, String er
         }
         serviceOrder.setTaskStatus(taskStatus);
         serviceOrder.setCompletedTime(OffsetDateTime.now());
-        serviceOrder.setErrorMsg(errorMsg);
+        serviceOrder.setError(ErrorResponse
```

Comment:
in line 112 - the argument type for errorMsg must be  ErrorResponse and let calling methods to define the correct exception.